drivers/ethernet/eth_stm32_hal.c

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "dev/eth_stm32_hal"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+#include <logging/sys_log.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <soc.h>
+#include <misc/printk.h>
+#include <clock_control.h>
+#include <clock_control/stm32_clock_control.h>
+
+#include <stm32f4xx_hal_eth.h>
+
+#include "eth_stm32_hal_priv.h"
+
+static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __aligned(4);
+static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __aligned(4);
+static u8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __aligned(4);
```

Comment:
This uses lot of memory. Would it be possible to do this same way as eth_sam_gmac.c is doing, so mapping dma region(s) directly to net_buf data area?